app/src/sharedTest/java/org/oppia/app/parser/FractionInteractionViewModelTest.kt

```diff
@@ -0,0 +1,155 @@
+    assertThat(numberWithUnits.fraction.denominator).isEqualTo(5)
+    assertThat(numberWithUnits.getUnit(0).unit).isEqualTo("km/hr")
+  }
+
+  @Test
+  fun testNumberWithUnitsInputInteractionView_withCurrency_hasCorrectPendingAnswer() {
+    val numberWithUnits = getPendingAnswer("Rs 10000").answer.numberWithUnits
+    assertThat(numberWithUnits.getUnit(0).unit).isEqualTo("Rs")
+    assertThat(numberWithUnits.real).isEqualTo(10000.0)
+  }
+
+  @Test
+  fun testNumberWithUnitsInputInteractionView_withFractionWithWholeNumberValueAndUnit_hasCorrectPendingAnswer() {
+    val numberWithUnits = getPendingAnswer("5 9/10 km").answer.numberWithUnits
+    assertThat(numberWithUnits.fraction.isNegative).isEqualTo(false)
+    assertThat(numberWithUnits.fraction.wholeNumber).isEqualTo(5)
+    assertThat(numberWithUnits.fraction.numerator).isEqualTo(9)
+    assertThat(numberWithUnits.fraction.denominator).isEqualTo(10)
+    assertThat(numberWithUnits.getUnit(0).unit).isEqualTo("km")
+  }
+
+  @Test
+  fun testNumberWithUnitsInputInteractionView_withCurrencyWithDot_hasCorrectPendingAnswer() {
+    val numberWithUnits = getPendingAnswer("100 Rs.").answer.numberWithUnits
+    assertThat(numberWithUnits.real).isEqualTo(100.0)
+    assertThat(numberWithUnits.getUnit(0).unit).isEqualTo("Rs.")
+  }
+
+  @Test
+  fun testNumberWithUnitsInputInteractionView_withDollarSymbol_hasCorrectPendingAnswer() {
+    val numberWithUnits = getPendingAnswer("$ 100").answer.numberWithUnits
+    assertThat(numberWithUnits.real).isEqualTo(100.0)
+    assertThat(numberWithUnits.getUnit(0).unit).isEqualTo("$")
+  }
+
+  @Test
+  fun testNumberWithUnitsInputInteractionView_withDecimalValueAndUnit_hasCorrectPendingAnswer() {
+    val numberWithUnits = getPendingAnswer("100.5 Km").answer.numberWithUnits
+    assertThat(numberWithUnits.real).isEqualTo(100.5)
+    assertThat(numberWithUnits.getUnit(0).unit).isEqualTo("Km")
+  }
+
+  @Test
+  fun testNumberWithUnitsInputInteractionView_withWithDays_hasCorrectPendingAnswer() {
+    val numberWithUnits = getPendingAnswer("3 1/2 days").answer.numberWithUnits
+    assertThat(numberWithUnits.fraction.wholeNumber).isEqualTo(3)
+    assertThat(numberWithUnits.fraction.numerator).isEqualTo(1)
+    assertThat(numberWithUnits.fraction.denominator).isEqualTo(2)
+    assertThat(numberWithUnits.getUnit(0).unit).isEqualTo("days")
+  }
+
+  @Test
+  fun testNumberWithUnitsInputInteractionView_withNegativeDecimal_hasCorrectPendingAnswer() {
+    val numberWithUnits = getPendingAnswer("-10.00 m").answer.numberWithUnits
+    assertThat(numberWithUnits.real).isEqualTo(-10.0)
+    assertThat(numberWithUnits.getUnit(0).unit).isEqualTo("m")
+  }
+
+  @Test
+  fun testNumberWithUnitsInputInteractionView_withInputtedWithSymbols_hasCorrectPendingAnswer() {
+    val numberWithUnits = getPendingAnswer("100.5 Km/m^2").answer.numberWithUnits
+    assertThat(numberWithUnits.real).isEqualTo(100.5)
+    assertThat(numberWithUnits.getUnit(0).unit).isEqualTo("Km/m^2")
```

Comment:
What about verifying the exponent? Shouldn't this be two units, one with Km exponent 1 and m with exponent -2?